### PR TITLE
[soft-assert] Fix missing attachments with enabled fail-fast at scenario level

### PR DESCRIPTION
**Pull request by draker94**

### CODE REVIEW

File: vividus-soft-assert/src/main/java/org/vividus/softassert/ISoftAssert.java [hunk 1]
```diff
@@ -31,6 +32,16 @@
      */
     boolean assertTrue(String description, boolean condition);
 
+    /**
+     * Asserts that a condition is true. If it isn't, an assertion error is added to the collection.
+     * Can be specified an additional operation which will always be executed.
```
Comment (valfirst) [inline vividus-soft-assert/src/main/java/org/vividus/softassert/ISoftAssert.java:37]:
```suggestion
     * An additional operation which will always be executed is specified.
```

Comment (valfirst) [reply vividus-soft-assert/src/main/java/org/vividus/softassert/ISoftAssert.java:37]: also please add the description the operation is extremely useful when fail-fast is enabled

File: vividus/src/main/java/org/vividus/steps/ArchiveSteps.java [hunk 2]
```diff
@@ -110,28 +112,41 @@ public void saveArchiveEntriesToVariables(DataWrapper archiveData, List<ArchiveV
...
-                    comparisonRule, expectedName), entryNames, hasItem(comparisonRule.createMatcher(expectedName)));
+                    comparisonRule, expectedName), entryNames, hasItem(comparisonRule.createMatcher(expectedName)),
+                    publishAttachmentOnFailure(entryNames, reportNotPublished));
         }
-        return softAssert.assertThat("The archive contains entry with name " + expectedName, entryNames,
-                hasItem(expectedName));
+        else
+        {
+            softAssert.assertThat("The archive contains entry with name " + expectedName, entryNames,
+                    hasItem(expectedName), publishAttachmentOnFailure(entryNames, reportNotPublished));
```
Comment (valfirst) [inline vividus/src/main/java/org/vividus/steps/ArchiveSteps.java:135]: it makes sense to refactor to have only one invocation of `softAssert`

File: vividus-soft-assert/src/main/java/org/vividus/softassert/ISoftAssert.java [hunk 3]
```diff
@@ -31,6 +32,16 @@
...
+     * @param description The assertion description
+     * @param condition Condition to be checked
+     * @param resultConsumer Always executed additional operation that consumes result of the assertion
```
Comment (valfirst) [inline vividus-soft-assert/src/main/java/org/vividus/softassert/ISoftAssert.java:40]: it seems it's used only when assertion is failed, does it make sense to convert it to something like: `Runnable failureHandler`?

**Review by valfirst — CHANGES_REQUESTED** [review]

File: vividus-soft-assert/src/main/java/org/vividus/softassert/ISoftAssert.java (hunk 3 above)
Comment (draker94) [reply vividus-soft-assert/src/main/java/org/vividus/softassert/ISoftAssert.java:40]: We have two places where we publish attachments unconditionally: [ResourceCheckSteps](https://github.com/vividus-framework/vividus/blob/66ff1945a8001219a735e61a39715a01178e414f/vividus-plugin-web-app-to-rest-api/src/main/java/org/vividus/steps/integration/ResourceCheckSteps.java#L128-L129) and [HttpResourceValidationSteps](https://github.com/vividus-framework/vividus/blob/6e1be1484f06d59151f7de0aadbe72ace2e5c20f/vividus-plugin-rest-api/src/main/java/org/vividus/steps/http/HttpResourceValidationSteps.java#L63-L64). These cases will be fixed in separate PR.

**Review by valfirst — APPROVED** [review]

**Review by ikalinin1 — APPROVED** [review]

**Review by EDbarvinsky — APPROVED** [review]

**Review by uarlouski — APPROVED** [review]

**Pull request merged**